### ISSUE TITLE
keepassxc not using touchid on macosx 12.1 beta

### ISSUE THREAD

**Issue by matthiasschroder** [keepassxreboot/keepassxc#7177]
## Overview
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Give a BRIEF summary about your problem )
Since I'm using macOS 12.1 beta keepassxc no longer uses the touchid for re-opening a database. So now I always have to use the masterpassword after a timeout has occurred.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. Upgrade to macOS 12.1 beta (not sure with which one this started)
2. Open database
3. Wait until timeout occurs
4. Try to re-open without password

## Expected Behavior
[NOTE]: # ( Tell us what you expected to happen )
I expect t get a dialog window telling me that Keepassxc wants to use TouchID

## Actual Behavior
[NOTE]: # ( Tell us what actually happens )
I get a dialog telling me that I tried to re-open a db without password

## Context
[NOTE]: # ( Give us any additional information you may have. )

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - VERSION
Revision: REVISION

KeePassXC - Version 2.6.6
Revision: 386b79a

Qt 5.15.2
Debugging mode is disabled.

Operating system: macOS 12.1
CPU architecture: x86_64
Kernel: darwin 21.2.0

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare (signed and unsigned sharing)
- YubiKey
- TouchID

Cryptographic libraries:
- libgcrypt 1.9.3

[NOTE]: # ( Pick choices based on your environment )
Operating System: macOS
Desktop Env: 
Windowing System:

**Comment by droidmonkey** [comment]
You'll have to use a non beta operating system to ensure it isn't a transient error with macos. Obviously nothing changed on our end. Check your security and privacy settings.

**Comment by matthiasschroder** [comment]
Hi Jonathan,

thanks for your feedback. My reason for reporting this was to make sure you guys can have a look at this and apply eventually needed corrections in time before 12.1 is being released...

Let me know whether I can help with logs or additional debug info.

Best regards

**Comment by droidmonkey** [comment]
Of course. I just checked the local authentication docs for macos and they was no mention of a major change to touch id in 12.1

**Comment by alexdohmen** [comment]
@droidmonkey having the same issue on Mac 12.0.1. Same architecture and version numbers.

**Comment by matthiasschroder** [comment]
Interesting observation: if I put my finger on the TouchID before clicking 'OK' on the keepassxc "open" dialog it offers to use TouchID.

**Comment by phoerious** [comment]
I tested it in 12.0.1 and it's working without issues there. I don't have the 12.1 beta to test.

**Comment by alexdohmen** [comment]
@phoerious what is your keepassxc version number?

--
EDIT
I tested it on intel based Mac Book and there it works. Same versions for all other stuff. Only difference: I use dark theme on the M1 MacBook.

**Comment by sebkr** [comment]
Same here, I can't use Touch ID to unlock the database for a few days.

---
KeePassXC 2.6.6
macOS 12.0.1
MacBook Air M1 2020

**Comment by phoerious** [comment]
TouchID on 12.0 should work. Try deleting ~/Library/Application Support/KeePassXC/keepassxc.ini.

**Comment by alexdohmen** [comment]
~~Hm, seems to be more complex. It works if I close it by hand, but I have trouble opening with TouchID after it closed automatically.~~

---
EDIT: ok, found my issue.

![image](https://user-images.githubusercontent.com/6403142/148509081-195287c5-6608-4f24-b0a1-ee13bd8af311.png)

This was by default enabled, you have to enter your password again after lock.

**Comment by lifepillar** [comment]
I have this issue under macOS 12.1, but only with a MacBook Pro M1. TouchID works as expected under macOS 12.1 on an Intel MacBook Pro. In both systems I have installed KeepassXC 2.6.6 and both are configured in the same way (including the preference above). The only difference is that on Apple Silicon I have installed KeepassXC using MacPorts, while on Intel I have used Homebrew.

This is what I do in both systems:

1. Open KeepassXC
2. Check “TouchID for Quick Unlock” if not checked
3. Type the passphrase and click OK
4. Lock the database.

Now, on Intel MacBook Pro “TouchID for Quick Unlock” remains checked, but on MacBook Pro M1 it reverts to unchecked state. In other words, that setting does not seem to stick.
